File: src/android/WannatalkCore.java
package cordova_plugin_wannatalk_core;


import android.app.Activity;
import android.os.Bundle;
import android.util.Log;

import org.apache.cordova.CordovaPlugin;
import org.apache.cordova.CallbackContext;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;

import wannatalk.wannatalksdk.WTCore.Interface.IWTCompletion;
import wannatalk.wannatalksdk.WTCore.Interface.IWTLoginManager;
import wannatalk.wannatalksdk.WTCore.WTSDKManager;
import wannatalk.wannatalksdk.WTLogin.WTLoginManager;


/**
 * This class echoes a string called from JavaScript.
 */
public class WannatalkCore extends CordovaPlugin {

@Override
    public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
        if (action.equals("coolMethod")) {
            String message = args.getString(0);
            this.coolMethod(message, callbackContext);
            return true;
        }
        else if (action.equals("invokeMethod")) {
          this.invokeMethod(args, callbackContext);
          return true;
        }
        else if (action.equals("initializeSDK")) {
          this.initializeSDK(args, callbackContext);
          return true;
        }

        return false;
    }

    private void coolMethod(String message, CallbackContext callbackContext) {
        if (message != null && message.length() > 0) {
            callbackContext.success(message);
        } else {
            callbackContext.error("Expected one non-empty string argument.");
        }
    }

  CallbackContext loginCallbackContext = null;
  CallbackContext logoutCallbackContext = null;
  CallbackContext orgProfileCallbackContext = null;

  private void invokeMethod(JSONArray args, CallbackContext callbackContext) {

    try {
//      "identifier" -> "+919000220455"
//      "method" -> "silentLogin"
      JSONObject jsonObject  = (JSONObject) args.get(0);
      String method = (String) jsonObject.get("method");

      switch (method) {
        case "silentLogin": {
          String identifier = (String) jsonObject.get("identifier");
          Map<String, String> userInfo = new HashMap<String, String>();
          userInfo.put("displayName", "SG Test");

          silentLogin(identifier, userInfo, callbackContext);

          break;
        }
        case "login": {

          login(callbackContext);

          break;
        }
        case "loadOrgProfile": {
          loadOrganizationProfile(true, callbackContext);
          break;
        }
        case "logout": {
          logout(callbackContext);
          break;
        }
        case "isUserLoggedIn": {
          Boolean loggedIn = isUserLoggedIn();
          callbackContext.success(loggedIn?1:0);
          break;
        }
        default: {
          callbackContext.error("Invalid method.");
        }
      }

    } catch (JSONException e) {
      e.printStackTrace();
      callbackContext.error("Something went wrong with input data");
    }

  }


  public Boolean isUserLoggedIn() {
    return WTLoginManager.IsUserLoggedIn();
  }


  private void initializeSDK(JSONArray args, CallbackContext callbackContext) throws JSONException  {

    WTSDKManager.ShowGuideButton(false);
    WTSDKManager.ShowProfileInfoPage(false);
    WTSDKManager.AllowAddParticipants(false);
    WTSDKManager.AllowSendAudioMessage(false);
    WTSDKManager.EnableAutoTickets(true);
    WTSDKManager.ShowWelcomeMessage(true);

    WTLoginManager.setIwtLoginManager(iwtLoginManager);


    WTSDKManager.ShowExitButton(true);
    WTSDKManager.EnableChatProfile(false);

    JSONObject jsonobj  = (JSONObject) args.get(0);

    if (jsonobj != null) {
      Iterator<String> keys = jsonobj.keys();
      while(keys.hasNext()) {
        String key = keys.next();
        int iKey = Integer.parseInt(key);
        Object value = jsonobj.get(key);
        HandleMethodType(iKey, value);

      }

    }

    callbackContext.success();
  }


  IWTLoginManager iwtLoginManager = new IWTLoginManager() {
    @Override
    public void wtsdkUserLoggedOut() {

      logoutCallbackContext.success();
    }

    @Override
    public void wtsdkUserLoggedIn() {

      loginCallbackContext.success();

    }

    @Override
    public void wtsdkUserLoginFailed(String s) {

      Log.d("Main Activity", "wtsdkUserLoginFailed " + s);
      loginCallbackContext.error(s);
    }

    @Override
    public void wtsdkUserLogoutFailed(String s) {

      Log.d("Main Activity", "wtsdkUserLogoutFailed: " + s);
      logoutCallbackContext.error(s);
    }
  };

  void login(CallbackContext result) {
    loginCallbackContext = result;
    // Silent authentication without otp verification

    Activity currentActivity = this.cordova.getActivity();// getActivity();
    Bundle bundle = new Bundle();
    WTLoginManager.StartLoginActivity(currentActivity);
  }

  void silentLogin(final String identifier, final Map<String, String> userInfo, CallbackContext result) {
    loginCallbackContext = result;
    // Silent authentication without otp verification

    Activity currentActivity = this.cordova.getActivity();// getActivity();
    Bundle bundle = new Bundle();
    if (userInfo != null) {
      for (Map.Entry<String,String> entry : userInfo.entrySet()) {
        String key = entry.getKey();
        String value = entry.getValue();
        bundle.putString(key, value);
      }
    }

    WTLoginManager.SilentLoginActivity(identifier, bundle, currentActivity);

  }


  void logout(CallbackContext result) {
    logoutCallbackContext = result;
    Activity currentActivity = this.cordova.getActivity(); // getActivity();
    WTLoginManager.Logout(currentActivity);

  }

  void loadOrganizationProfile(Boolean autoOpenChat,final CallbackContext result) {
    // Load organization profile
    Activity currentActivity = this.cordova.getActivity();//getActivity();
    WTSDKManager.LoadOrganizationActivity(currentActivity, autoOpenChat, new IWTCompletion() {
      @Override
      public void onCompletion(boolean success, String error) {
        if (success) {
          if (result != null) {
            result.success();
          }
        }
        else {
          if (result != null) {
            result.error(error);
          }
        }

      }
    });

  }

  private void sendLoginCallback(String error) {
    if (loginCallbackContext != null) {
      if (error == null) {
        loginCallbackContext.success();
      }
      else {
        loginCallbackContext.error(error);
      }
    }

    loginCallbackContext = null;
  }

  private void sendLogoutCallback(String error) {
    if (logoutCallbackContext != null) {
      if (error == null) {
        logoutCallbackContext.success();
      }
      else {
        logoutCallbackContext.error(error);
      }
    }

    logoutCallbackContext = null;
  }


  // @Override
    // public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
    //     if (action.equals("coolMethod")) {
    //         String message = args.getString(0);
    //         this.coolMethod(message, callbackContext);
    //         return true;
    //     }
    //     return false;
    // }

    // private void coolMethod(String message, CallbackContext callbackContext) {
    //     if (message != null && message.length() > 0) {
    //         callbackContext.success(message);
    //     } else {
    //         callbackContext.error("Expected one non-empty string argument.");
    //     }
    // }

  static final int _kWTConfigClearTempFolder = 9001;
  static final int _kWTConfigSetInactiveChatTimeout = 9002;
  static final int _kWTConfigShowGuideButton = 9003;
  static final int _kWTConfigAllowSendAudioMessage = 9004;
  static final int _kWTConfigAllowAddParticipant = 9005;
  static final int _kWTConfigAllowRemoveParticipants = 9006;
  static final int _kWTConfigShowWelcomePage = 9007;
  static final int _kWTConfigShowProfileInfoPage = 9008;
  static final int _kWTConfigEnableAutoTickets = 9009;
  static final int _kWTConfigShowExitButton = 9010;
  static final int _kWTConfigShowChatParticipants = 9011;
  static final int _kWTConfigEnableChatProfile = 9012;
  static final int _kWTConfigAllowModifyChatProfile = 9013;
  static final int _kWTConfigSetAgentQueueInterval = 9014;

  public static void HandleMethodType(int methodType, Object arg) {

    switch (methodType) {
      case _kWTConfigClearTempFolder: {
        ClearTempFiles();
        break;
      }
      case _kWTConfigSetInactiveChatTimeout: {
        long interval = (long) arg;
        SetInactiveChatTimeoutInterval(interval);
        break;
      }
      case _kWTConfigSetAgentQueueInterval: {
        long interval = (long) arg;
        SetAgentQueueInterval(interval);
        break;
      }
      case _kWTConfigShowGuideButton: {
        boolean show = (boolean) arg;
        ShowGuideButton(show);
        break;
      }
      case _kWTConfigAllowSendAudioMessage: {
        boolean allow = (boolean) arg;
        AllowSendAudioMessage(allow);
        break;
      }
      case _kWTConfigAllowAddParticipant: {
        boolean allow = (boolean) arg;
        AllowAddParticipants(allow);
        break;
      }
      case _kWTConfigAllowRemoveParticipants: {
        boolean allow = (boolean) arg;
        AllowRemoveParticipants(allow);
        break;
      }
      case _kWTConfigShowWelcomePage: {
        boolean show = (boolean) arg;
        ShowWelcomeMessage(show);
        break;
      }
      case _kWTConfigShowProfileInfoPage: {
        boolean show = (boolean) arg;
        ShowProfileInfoPage(show);
        break;
      }
      case _kWTConfigEnableAutoTickets: {
        boolean enable = (boolean) arg;
        EnableAutoTickets(enable);
        break;
      }
      case _kWTConfigShowExitButton: {
        boolean show = (boolean) arg;
        ShowExitButton(show);
        break;
      }
      case _kWTConfigShowChatParticipants: {
        boolean show = (boolean) arg;
        ShowChatParticipants(show);
        break;
      }
      case _kWTConfigEnableChatProfile: {
        boolean enable = (boolean) arg;
        EnableChatProfile(enable);
        break;
      }
      case _kWTConfigAllowModifyChatProfile: {
        boolean allow = (boolean) arg;
        AllowModifyChatProfile(allow);
        break;
      }
      default: {

        break;
      }

    }
  }


  private static void ClearTempFiles() {
    WTSDKManager.ClearOldTempFiles();
    WTSDKManager.ClearTempFiles();
  }


  private static void ShowGuideButton(boolean show) {
    WTSDKManager.ShowGuideButton(show);
  }


  private static void ShowProfileInfoPage(boolean show) {
    WTSDKManager.ShowProfileInfoPage(show);
  }


  private static void AllowSendAudioMessage(boolean allow) {
    WTSDKManager.AllowSendAudioMessage(allow);
  }


  private static void AllowAddParticipants(boolean allow) {
    WTSDKManager.AllowAddParticipants(allow);
  }


  private static void AllowRemoveParticipants(boolean allow) {
    WTSDKManager.AllowRemoveParticipants(allow);
  }


  private static void ShowWelcomeMessage(boolean show) {
    WTSDKManager.ShowWelcomeMessage(show);
  }


  private static void EnableAutoTickets(boolean enable) {
    WTSDKManager.EnableAutoTickets(enable);
  }


  private static void ShowExitButton(boolean show) {
    WTSDKManager.ShowExitButton(show);
  }


  private static void ShowChatParticipants(boolean show) {
    WTSDKManager.ShowChatParticipants(show);
  }


  private static void EnableChatProfile(boolean enable) {
    WTSDKManager.EnableChatProfile(enable);
  }


  private static void AllowModifyChatProfile(boolean allow) {
    WTSDKManager.AllowModifyChatProfile(allow);
  }


  private static void SetInactiveChatTimeoutInterval(long timeoutInSeconds) {
    WTSDKManager.SetInactiveChatTimeoutInterval(timeoutInSeconds);
  }

  private static void SetAgentQueueInterval(long timeIntervalInSeconds) {
    WTSDKManager.SetAgentQueueInterval(timeIntervalInSeconds);
  }

}

File: src/ios/WannatalkCore.m
/********* WannatalkCore.m Cordova Plugin Implementation *******/

#import <Cordova/CDV.h>
//#import "WTConfigHandler.h"

#import <WTExternalSDK/WTExternalSDK.h>

@interface WannatalkCore : CDVPlugin {
  // Member variables go here.
    
}

@property BOOL tempBool;

@property BOOL sdkInitialized;

- (void)initializeSDK:(CDVInvokedUrlCommand*)command;
- (void)coolMethod:(CDVInvokedUrlCommand*)command;
- (void)invokeMethod:(CDVInvokedUrlCommand*)command;

@property CDVInvokedUrlCommand *loginResult;
@property CDVInvokedUrlCommand *logoutResult;
@property CDVInvokedUrlCommand *orgProfileResult;

@end

@interface WannatalkCore()<WTLoginManagerDelegate, WTSDKManagerDelegate>

@end

@implementation WannatalkCore
//
//- (void)coolMethod:(CDVInvokedUrlCommand*)command
//{
//    CDVPluginResult* pluginResult = nil;
//    NSString* echo = [command.arguments objectAtIndex:0];
//
//    if (echo != nil && [echo length] > 0) {
//        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:echo];
//    } else {
//        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
//    }
//
//    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
//}


- (void)coolMethod:(CDVInvokedUrlCommand*)command
{
    NSLog(@"coolMethod: %@", command);
    CDVPluginResult* pluginResult = nil;
    NSString* echo = [command.arguments objectAtIndex:0];

    if (echo != nil && [echo length] > 0) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:echo];
    } else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
    }
    
    pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

//
//- (void)isUserLoggedIn:(CDVInvokedUrlCommand*)command
//{
//    NSLog(@"isUserLoggedIn: %@", command);
//    BOOL loggedIn = [self isUserLoggedIn];
//
//    CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsBool:loggedIn];
//
//    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
//}


- (void)invokeMethod:(CDVInvokedUrlCommand*)command
{
//    NSLog(@"invokeMethod: %@", command);
    
    
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{

        dispatch_async(dispatch_get_main_queue(), ^{
            
//            if (!self.sdkInitialized) {
//                [self initializeSDKSettings:nil];
//            }
            
            CDVPluginResult* pluginResult = nil;
            NSDictionary* echo = [command.arguments objectAtIndex:0];

            
            NSString *method = echo[@"method"];
            
            if ([method isEqualToString:@"silentLogin"]) {
                
                NSString *identifier = echo[@"identifier"];
                NSDictionary *userInfo = echo[@"userInfo"];
                NSLog(@"isMainThread: %d", [NSThread isMainThread]);
                [self silentLogin:identifier userInfo:userInfo result:command];

                return;
            }
            else if ([method isEqualToString:@"login"]) {
                
                [self loginWithResult:command];

                return;
            }
            else if ([method isEqualToString:@"loadOrgProfile"]) {
            
                [self loadOrganizationProfile:YES result:command];
                return;
            }
            else if ([method isEqualToString:@"logout"]) {
            
                [self logout:command];
                return;
            }
//            else if ([method isEqualToString:@"updateConfig"]) {
//
//                NSInteger methodType = [echo[@"methodType"] integerValue];
//                NSNumber *args = echo[@"arg"];
//                [WannatalkCore handleConfigMethodType:methodType arguments:args];
//                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
//                [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
//                return;
//            }
            else if ([method isEqualToString:@"isUserLoggedIn"]) {
                BOOL loggedIn = [self isUserLoggedIn];
                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsBool:loggedIn];
                [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
                return;
            }
            
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
            [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
        });

    });


}


- (void)initializeSDK:(CDVInvokedUrlCommand*)command
{
    NSLog(@"initializeSDK: %@", command);
    
    self.sdkInitialized = YES;
    
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
        
        dispatch_async(dispatch_get_main_queue(), ^{

            NSDictionary *sdkSettings = [command.arguments objectAtIndex:0];
            
            [self initializeSDKSettings:sdkSettings];
            
            CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
            [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];

            
        });
        
    });

}

- (void) initializeSDKSettings:(NSDictionary *) sdkSettings {
    
    UIApplication *application =  [UIApplication sharedApplication];
    NSDictionary *launchOptions = nil;
    
    [[WTSDKApplicationDelegate sharedInstance] application:application didFinishLaunchingWithOptions:launchOptions];
    [WTLoginManager sharedInstance].delegate = self;
    
    if (sdkSettings) {
        
//        if ([sdkSettings )
        
//        NSDictionary<NSNumber *, NSNumber *> *dctTest = @{};
        [WTSDKManager ShowGuideButton:NO];
        [WTSDKManager ShowProfileInfoPage:NO];
        [WTSDKManager ShouldAllowSendAudioMessage:NO];
        [WTSDKManager ShouldAllowAddParticipant:NO];
        [WTSDKManager EnableAutoTickets:YES];

        [WTSDKManager ShowExitButton:YES];
        [WTSDKManager EnableChatProfile:NO];
        
        [sdkSettings enumerateKeysAndObjectsUsingBlock:^(NSNumber * _Nonnull key, NSNumber * _Nonnull obj, BOOL * _Nonnull stop) {
           
            [WannatalkCore handleConfigMethodType:key.integerValue arguments:obj];
            
        }];
        
        
    }
    
}



//- (IBAction)silentLoginBtnClicked:(id)sender {
//    NSString *sad = @"";
//
//
//    @try {
//        NSDictionary *userInfo = @{ @"key1": @"value1", @"key2": sad};
//
//        [[WTLoginManager sharedInstance] silentLoginWithIdentifier:@"+919000220455" userInfo:userInfo fromVC:self];
//    } @catch (NSException *exception) {
//        NSLog(@"Exception: %@", exception);
//    }
//
//
//}


- (void) sendLoginCallback:(NSString *) error {
    
    
    [self.commandDelegate runInBackground:^{
        
        if (self.loginResult) {
            CDVPluginResult *pluginResult;
            
            if (error) {
                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:error];
            }
            else {
                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
            }
        
            [self.commandDelegate sendPluginResult:pluginResult callbackId:self.loginResult.callbackId];
            
        }
        self.loginResult = nil;
    }];
    
}


- (void) sendLogoutCallback:(NSString *) error {
    
    
    [self.commandDelegate runInBackground:^{
        
        
        if (self.logoutResult) {
            CDVPluginResult *pluginResult;
            
            if (error) {
                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:error];
            }
            else {
                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
            }
            
            
            [self.commandDelegate sendPluginResult:pluginResult callbackId:self.logoutResult.callbackId];
            
        }
        self.logoutResult = nil;
    }];
}



- (void) sendOrgProfileCallback:(NSString *) error {
    
    [self.commandDelegate runInBackground:^{
        
        if (self.orgProfileResult) {
            CDVPluginResult *pluginResult;
            
            //        if (echo != nil && [echo length] > 0) {
            //            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:echo];
            //        } else {
            //            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR];
            //        }
            
            if (error) {
                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:error];
            }
            else {
                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
            }
            
            //        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus];
            
            [self.commandDelegate sendPluginResult:pluginResult callbackId:self.orgProfileResult.callbackId];
            
        }
        self.orgProfileResult = nil;
    }];
       
}


#pragma mark -
// If implemented, this method will be invoked when organization profile loads successfully
- (void) wtsdkOrgProfileDidLoadSuccesfully {
    [self sendOrgProfileCallback:nil];
}

// If implemented, this method will be invoked when organization profile fails to load
- (void) wtsdkOrgProfileDidLoadFailWithError:(NSString *) error {
    [self sendOrgProfileCallback:error];
}



#pragma mark -

// This method will be invoked when user sign in successfully
- (void) wtAccountDidLoginSuccessfully {
    [self sendLoginCallback:nil];
}

// If implemented, this method will be invoked when login fails
- (void) wtAccountDidLoginFailWithError:(NSString *) error {
    [self sendLoginCallback:error];
}

// This method will be invoked when user sign out successfully
- (void) wtAccountDidLogOutSuccessfully {
    [self sendLogoutCallback:nil];
}

// If implemented, this method will be invoked when logout fails
- (void) wtAccountDidLogOutFailedWithError:(NSString *) error {
    [self sendLogoutCallback:error];
}


#pragma mark -

+ (UIViewController *) GetWindowRootViewController {
    UIWindow *window = [[UIApplication sharedApplication] keyWindow];
    UIViewController *_rootViewController = (UIViewController *)window.rootViewController;
    return _rootViewController;
}

+ (void) RunMainThread:(void (^)(void))onMainThread {

    if (onMainThread) {
        onMainThread();
    }
//
//    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
//        dispatch_async(dispatch_get_main_queue(), ^{
//            if (onMainThread) {
//                onMainThread();
//            }
//        });
//    });

}

+ (void) GetBaseViewController:(void (^)(UIViewController *baseViewController)) onCompletion {

    [self RunMainThread:^{
        UIViewController *_rootViewController = [self GetWindowRootViewController];

        if (onCompletion) {
            onCompletion(_rootViewController);
        }
    }];


}
#pragma mark -

- (BOOL) isUserLoggedIn {
    return [WTLoginManager sharedInstance].isUserLoggedIn;
}

- (void) logout:(CDVInvokedUrlCommand *) command {
    self.logoutResult = command;

    [WannatalkCore RunMainThread:^{
        [[WTLoginManager sharedInstance] logout];
    }];
}


- (void)loginWithResult:(CDVInvokedUrlCommand *) command {
    self.loginResult = command;

    [WannatalkCore GetBaseViewController:^(UIViewController *baseViewController) {
        [[WTLoginManager sharedInstance] loginFromVC:baseViewController];
    }];
}

- (void)silentLogin:(NSString *) identifier userInfo:(NSDictionary *) userInfo result:(CDVInvokedUrlCommand *) command {
    self.loginResult = command;

    
    [WannatalkCore GetBaseViewController:^(UIViewController *baseViewController) {
        [[WTLoginManager sharedInstance] silentLoginWithIdentifier:identifier userInfo:userInfo fromVC:baseViewController];

    }];
}


-(void) login:(CDVInvokedUrlCommand *) command {

    self.loginResult = command;

    [WannatalkCore GetBaseViewController:^(UIViewController *baseViewController) {
        [[WTLoginManager sharedInstance] loginFromVC:baseViewController];

    }];

}

#pragma mark -

- (void)loadOrganizationProfile:(BOOL) autoOpenChat result:(CDVInvokedUrlCommand *) command {
    self.orgProfileResult = command;

    [WannatalkCore GetBaseViewController:^(UIViewController *baseViewController) {
        [baseViewController presentOrgProfileVCWithAutoOpenChat:autoOpenChat delegate:self animated:YES completion:nil];
    }];

}

-(void) loadChatList:(CDVInvokedUrlCommand *) command {
//    self.chatListResult = result;
//    [WannatalkCore GetBaseViewController:^(UIViewController *baseViewController) {
//        [baseViewController presentChatListVCWithDelegate:self animated:YES completion:nil];
//    }];
}


- (void)loadUsers:(CDVInvokedUrlCommand *) command {
//    self.userListResult = result;
//
//    [WannatalkCore GetBaseViewController:^(UIViewController *baseViewController) {
//        [baseViewController presentUsersVCWithDelegate:self animated:YES completion:nil];
//    }];
}

#pragma mark -

static const  int _kWTConfigClearTempFolder = 9001;
static const  int _kWTConfigSetInactiveChatTimeout = 9002;
static const  int _kWTConfigShowGuideButton = 9003;
static const  int _kWTConfigAllowSendAudioMessage = 9004;
static const  int _kWTConfigAllowAddParticipant = 9005;
static const  int _kWTConfigAllowRemoveParticipants = 9006;
static const  int _kWTConfigShowWelcomePage = 9007;
static const  int _kWTConfigShowProfileInfoPage = 9008;
static const  int _kWTConfigEnableAutoTickets = 9009;
static const  int _kWTConfigShowExitButton = 9010;
static const  int _kWTConfigShowChatParticipants = 9011;
static const  int _kWTConfigEnableChatProfile = 9012;
static const  int _kWTConfigAllowModifyChatProfile = 9013;
static const  int _kWTConfigSetAgentQueueInterval = 9014;

#define cShow @"show"
#define cEnable @"enable"
#define cAllow @"allow"

#define cTimeoutInterval @"timeoutInterval"
#define cTimeInterval @"timeInterval"

+ (void) handleConfigMethodType:(NSInteger) methodType arguments:(NSNumber *) arg {

    switch (methodType) {
        case _kWTConfigClearTempFolder: {
            [self ClearTempFiles];
            break;
        }
        case _kWTConfigSetInactiveChatTimeout: {
            NSTimeInterval interval = [arg doubleValue];
            [self SetInactiveChatTimeoutInterval:interval];
            break;
        }
        case _kWTConfigSetAgentQueueInterval: {
            NSTimeInterval interval = [arg doubleValue];
            [self SetAgentQueueTimeInterval:interval];
            break;
        }
        case _kWTConfigShowGuideButton: {
            BOOL show = [arg boolValue];
            [self ShowGuideButton:show];
            break;
        }
        case _kWTConfigAllowSendAudioMessage: {
            BOOL allow = [arg boolValue];
            [self AllowSendAudioMessage:allow];
            break;
        }
        case _kWTConfigAllowAddParticipant: {
            BOOL allow = [arg boolValue];
            [self AllowAddParticipants:allow];
            break;
        }
        case _kWTConfigAllowRemoveParticipants: {
            BOOL allow = [arg boolValue];
            [self AllowRemoveParticipants:allow];
            break;
        }
        case _kWTConfigShowWelcomePage: {
            BOOL show = [arg boolValue];
            [self ShowWelcomeMessage:show];
            break;
        }
        case _kWTConfigShowProfileInfoPage: {
            BOOL show = [arg boolValue];
            [self ShowProfileInfoPage:show];
            break;
        }
        case _kWTConfigEnableAutoTickets: {
            BOOL enable = [arg boolValue];
            [self EnableAutoTickets:enable];
            break;
        }
        case _kWTConfigShowExitButton: {
            BOOL show = [arg boolValue];
            [self ShowExitButton:show];
            break;
        }
        case _kWTConfigShowChatParticipants: {
            BOOL show = [arg boolValue];
            [self ShowChatParticipants:show];
            break;
        }
        case _kWTConfigEnableChatProfile: {
            BOOL enable = [arg boolValue];
            [self EnableChatProfile:enable];
            break;
        }
        case _kWTConfigAllowModifyChatProfile: {
            BOOL allow = [arg boolValue];
            [self AllowModifyChatProfile:allow];
            break;
        }
        default: {

            break;
        }

    }

}


#pragma mark -

+ (void)ClearTempFiles {
    [WTSDKManager ClearTempDirectory];
}

// To show or hide guide button
+ (void)ShowGuideButton:(BOOL) show               // default = YES
{
    [WTSDKManager ShowGuideButton:show];
}

// To enable or disable sending audio message
+ (void)AllowSendAudioMessage:(BOOL) allow  // default = YES
{
    [WTSDKManager ShouldAllowSendAudioMessage:allow];
}

// To show or hide add participants option in new ticket page and chat item profile page
+ (void)AllowAddParticipants:(BOOL) allow    // default = YES
{
    [WTSDKManager ShouldAllowAddParticipant:allow];
}

// To show or hide remove participants option in chat item profile
+ (void)AllowRemoveParticipants:(BOOL) allow // default = NO
{
    [WTSDKManager ShouldAllowRemoveParticipant:allow];
}

// To show or hide welcome message
+ (void)ShowWelcomeMessage:(BOOL) show            // default = NO
{
    [WTSDKManager ShowWelcomeMessage:show];
}

// To show or hide Profile Info page
+ (void)ShowProfileInfoPage:(BOOL) show           // default = YES
{
    [WTSDKManager ShowProfileInfoPage:show];
}

// To create auto tickets:
//Chat ticket will create automatically when auto tickets is enabled, otherwise default ticket creation page will popup
+ (void)EnableAutoTickets:(BOOL) enable           // default = NO
{
    [WTSDKManager EnableAutoTickets:enable];
}

// To show or hide close chat button in chat page
+ (void)ShowExitButton:(BOOL) show                // default = NO
{
    [WTSDKManager ShowExitButton:show];
}

// To show or hide participants in chat profile page
+ (void)ShowChatParticipants:(BOOL) show          // default = YES
{
    [WTSDKManager ShowChatParticipants:show];
}

// To enable or disbale chat profile page
+ (void)EnableChatProfile:(BOOL) enable           // default = YES
{
    [WTSDKManager EnableChatProfile:enable];
}

// To allow modify  in chat profile page
+ (void)AllowModifyChatProfile:(BOOL) allow       // default = YES
{
    [WTSDKManager AllowModifyChatProfile:allow];
}

// To set Inactive chat timeout:
//Chat session will end if user is inactive for timeout interval duration. If timeout interval is 0, chat session will not end automatically. The default timout interval is 1800 seconds (30 minutes).
+ (void)SetInactiveChatTimeoutInterval:(double) timeoutInterval   // default = 1800 seconds (30 minutes).
{
    [WTSDKManager SetInactiveChatTimeoutInterval:timeoutInterval];
}

// To set timeinterval for checking the available agent // default 20 seconds
+ (void) SetAgentQueueTimeInterval:(double) timeInterval {
    [WTSDKManager SetAgentQueueTimeInterval:timeInterval];
}


@end
